docs/docsite/rst/playbook_guide/playbooks_reuse_roles.rst

```diff
@@ -108,15 +124,21 @@ The classic (original) way to use roles is with the ``roles`` option for a given
         - common
         - webservers
 
-When you use the ``roles`` option at the play level, for each role 'x':
+When you use the ``roles`` option at the play level, each role 'x' looks for a ``main.yml/main.yaml/main`` in the following directories:
 
-- If roles/x/tasks/main.yml exists, Ansible adds the tasks in that file to the play.
-- If roles/x/handlers/main.yml exists, Ansible adds the handlers in that file to the play.
-- If roles/x/vars/main.yml exists, Ansible adds the variables in that file to the play.
-- If roles/x/defaults/main.yml exists, Ansible adds the variables in that file to the play.
-- If roles/x/meta/main.yml exists, Ansible adds any role dependencies in that file to the list of roles.
+- ``roles/x/tasks/``
+- ``roles/x/handlers/``
+- ``roles/x/vars/``
+- ``roles/x/defaults/``
+- ``roles/x/meta/``
 - Any copy, script, template or include tasks (in the role) can reference files in roles/x/{files,templates,tasks}/ (dir depends on task) without having to path them relatively or absolutely.
 
+.. note::
+    ``vars`` and ``defaults`` can also match to a directory of the same name and Ansible will process all the files contained in that directory. See :ref:`Role directory structure <role_directory_structure>` for more details.
+
+.. note::
+    If you use ``include_role/import_role``, you can specify a custom file name instead of ``main``. The ``meta`` directory is an exception because it does not allow for customization.
+
```

Comment:
It's unclear to me if this note applies to ALL main.yml, or only the ``vars/`` and ``defaults`` dir. @bcoca - do you know the answer here?

Reply:
it applies to tasks, handlers, vars and defaults, i would just link to improt_role/include_role docs where the `from_X` options are detailed